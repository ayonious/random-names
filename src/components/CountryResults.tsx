import * as React from "react";
import Paper from "@material-ui/core/Paper";
import { database } from "../resources/Database";
import { CustomButton } from "./Utils";

interface Props {
  country: string;
}

interface State {
  currentName: string;
}

export default class CountryReults extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.handleShuffle = this.handleShuffle.bind(this);
    this.state = { currentName: "" };
  }

  handleShuffle() {
    const index = Math.floor(
      Math.random() * Object.keys(database[this.props.country].names).length
    );
    const name = database[this.props.country].names[index];
    this.setState({ currentName: name });
  }

  render() {
    const style = {
      height: 50,
      width: 150,
      margin: "0 auto",
      fontSize: "xx-large",
      contentAlign: "center"
    };
    return (
      <div>
        <br />
        <div style={style}>
          <Paper square={true}>{this.state.currentName}</Paper>
        </div>
        <br />
        <div>
          <CustomButton onClick={this.handleShuffle} title={"Shuffle"} />{" "}
        </div>
      </div>
    );
  }
}
