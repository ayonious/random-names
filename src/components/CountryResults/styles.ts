import styled from 'styled-components';

export const CardWrapper = styled.div`
  height: 50px;
  width: 170px;
  margin: 0 auto;
  font-size: xx-large;
  text-align: center;
`;
