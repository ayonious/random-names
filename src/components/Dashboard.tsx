import * as React from "react";
import CountryReults from "./CountryResults";
import Select from "@material-ui/core/Select";
import MenuItem from "@material-ui/core/MenuItem";
import FormHelperText from "@material-ui/core/FormHelperText";
import FormControl from "@material-ui/core/FormControl";
import { database } from "../resources/Database";
const regions = Object.keys(database);

interface Props {}

interface State {
  selectedCountry: string;
  currentName: string;
}

export default class Dashboard extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.onSelectOption = this.onSelectOption.bind(this);
    this.state = { selectedCountry: "Germany", currentName: "" };
  }

  onSelectOption(event: any) {
    this.setState({ selectedCountry: event.target.value });
  }

  render() {
    const selectList = regions.map(region => (
      <MenuItem key={region} value={region}>
        {database[region].flag} {region}
      </MenuItem>
    ));
    const style = {
      display: "flex",
      width: 160,
      margin: "0 auto",
      alignContent: "center",
      justifyContent: "center"
    };

    return (
      <div>
        <FormControl style={style}>
          <Select
            value={this.state.selectedCountry}
            onChange={this.onSelectOption}
            autoWidth={false}
          >
            {selectList}
          </Select>
          <FormHelperText>Region</FormHelperText>
        </FormControl>
        <CountryReults country={this.state.selectedCountry} />
      </div>
    );
  }
}
