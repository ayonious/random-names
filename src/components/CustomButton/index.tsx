import Button from '@material-ui/core/Button';
import React from 'react';

import { CenteredFlexDevWrapper } from './styles';

interface Props {
  title: string;
  onClick: () => void;
}

const CustomButton = (props: Props) => {
  return (
    <CenteredFlexDevWrapper>
      <Button variant="contained" color="primary" onClick={props.onClick}>
        {props.title}
      </Button>
    </CenteredFlexDevWrapper>
  );
};

export default CustomButton;
