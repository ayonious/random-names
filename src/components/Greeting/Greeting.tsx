import React from 'react';

import { CenteredFlexDevWrapper } from './styles';

const Greeting = () => {
  return (
    <CenteredFlexDevWrapper>
      <h1> Start Shuffling! </h1>
    </CenteredFlexDevWrapper>
  );
};

export default Greeting;
