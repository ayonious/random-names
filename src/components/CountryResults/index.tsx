import Card from '@material-ui/core/Card';
import React, { useState } from 'react';

import { database } from '../../resources/Database';
import { CardWrapper } from './styles';
import CustomButton from '../CustomButton';

interface Props {
  country: string;
}

const CountryResults = (props: Props) => {
  const [currentName, changeName] = useState('');

  const handleShuffle = () => {
    const index = Math.floor(
      Math.random() * Object.keys(database[props.country].names).length
    );
    const name = database[props.country].names[index];
    changeName(name);
  };

  return (
    <div>
      <br />
      <CardWrapper>
        <Card square={true}> {currentName} </Card>
      </CardWrapper>
      <br />
      <div>
        <CustomButton onClick={handleShuffle} title={'Shuffle'} />{' '}
      </div>
    </div>
  );
};

export default CountryResults;
