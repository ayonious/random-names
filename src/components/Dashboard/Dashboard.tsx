import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import React, { useState } from 'react';

import { database } from '../../resources/Database';
import { FormWrapper } from './styles';
import CountryReults from '../CountryResults/CountryResults';
import Greeting from '../Greeting/Greeting';

const regions = Object.keys(database);

interface Props {}

interface State {
  selectedCountry: string;
}
const Dashboard = () => {
  const [selectedCountry, changeSelectedCountry] = useState('Germany');

  const onSelectOption = (event: any) => {
    changeSelectedCountry(event.target.value);
  };

  const selectList = regions.map((region) => (
    <MenuItem key={region} value={region}>
      {database[region].flag} {region}
    </MenuItem>
  ));
  return (
    <div>
      <div>
        <Greeting />
      </div>
      <FormWrapper>
        <FormControl className="myFormClass">
          <Select
            value={selectedCountry}
            onChange={onSelectOption}
            autoWidth={false}
          >
            {selectList}
          </Select>
          <FormHelperText>Region</FormHelperText>
        </FormControl>
        <CountryReults country={selectedCountry} />
      </FormWrapper>
    </div>
  );
};

export default Dashboard;
