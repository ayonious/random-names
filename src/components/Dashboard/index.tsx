import React, { useState } from 'react';
import { SingleValue } from 'react-select';

import { database } from '../../resources/Database';
import CountryResults from '../CountryResults';
import GithubCornerPart from '../GithubCornerPart';
import Greeting from '../Greeting';
import SelectCountry, { SelectItem } from '../SelectCountry';
import { PageWrapper } from './styles';

const regions = Object.keys(database);

const Dashboard = () => {
  const [selectedCountry, changeSelectedCountry] = useState<string>('Germany');

  const onSelectOption = (event: SingleValue<SelectItem>) => {
    changeSelectedCountry(event?.value || 'Germany');
  };

  const selectList: SelectItem[] = regions.map((region) => ({
    label: `${database[region].flag} ${region}`,
    value: region,
  }));

  return (
    <PageWrapper>
      <Greeting />
      <SelectCountry selectList={selectList} onSelectOption={onSelectOption} />
      <CountryResults country={selectedCountry} />
      <GithubCornerPart />
    </PageWrapper>
  );
};

export default Dashboard;
