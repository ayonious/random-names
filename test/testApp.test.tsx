/**
 * @jest-environment jsdom
 */

import React from 'react';
import expect from 'expect';

import { render } from '@testing-library/react';
import CountryResults from '../src/components/CountryResults';
import Dashboard from '../src/components/Dashboard';
import Greeting from '../src/components/Greeting';
import CustomButton from '../src/components/CustomButton';

it('CountryResults renders', () => {
  const { container } = render(<CountryResults country="Germany" />);
  expect(container).toMatchSnapshot();
});

it('Dashboard renders', () => {
  const { container } = render(<Dashboard />);
  expect(container).toMatchSnapshot();
});

it('CustomButton renders', () => {
  const func = () => {
    console.log('on click function testing');
  };

  const { container } = render(
    <CustomButton onClick={func} title={'Shuffle'} />
  );
  expect(container).toMatchSnapshot();
});

it('Greeting renders', () => {
  const { container } = render(<Greeting />);
  expect(container).toMatchSnapshot();
});
