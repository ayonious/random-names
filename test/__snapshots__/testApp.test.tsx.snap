// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryResults renders 1`] = `
<div>
  <div>
    <br />
    <div
      class="sc-bczRLJ eyXzyt"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1"
      >
         
         
      </div>
    </div>
    <br />
    <div>
      <div
        class="sc-gsnTZi hcNodn"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Shuffle
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
       
    </div>
  </div>
</div>
`;

exports[`CustomButton renders 1`] = `
<div>
  <div
    class="sc-gsnTZi hcNodn"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Shuffle
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Dashboard renders 1`] = `
<div>
  <div
    class="sc-iBkjds zVHAC"
  >
    <div
      class="sc-dkzDqf gcKDBg"
    >
      <h1>
         Start Shuffling! 
      </h1>
    </div>
    <div
      class="sc-hKMtZM"
    >
      <div
        class="sc-gKXOVf fCgQRH"
      >
        <div
          class="sc-eCYdqJ gTsWhs"
        >
          <div
            class=" css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-1s2u09g-control"
            >
              <div
                class=" css-319lph-ValueContainer"
              >
                <div
                  class=" css-qc6sy-singleValue"
                >
                  ðŸ‡©ðŸ‡ª Germany
                </div>
                <div
                  class=" css-6j8wv5-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-2-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="country list"
              type="hidden"
              value="Germany"
            />
          </div>
        </div>
        <div
          class="sc-jSMfEi chTvnt"
        >
          Region
        </div>
      </div>
      <div>
        <br />
        <div
          class="sc-bczRLJ eyXzyt"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1"
          >
             
             
          </div>
        </div>
        <br />
        <div>
          <div
            class="sc-gsnTZi hcNodn"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Shuffle
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
           
        </div>
      </div>
    </div>
    <a
      aria-label="Open GitHub project"
      class="github-corner"
      href="https://github.com/ayonious/random-names"
    >
      <svg
        height="80"
        style="position: absolute; top: 0px; fill: #fff; right: 0px;"
        viewBox="0 0 250 250"
        width="80"
      >
        <path
          class="octo-banner"
          d="M0 0l115 115h15l12 27 108 108V0z"
          fill="#5f32b3"
        />
        <path
          class="octo-arm"
          d="M128 109c-15-9-9-19-9-19 3-7 2-11 2-11-1-7 3-2 3-2 4 5 2 11 2 11-3 10 5 15 9 16"
          style="transform-origin: 130px 106px;"
        />
        <path
          class="octo-body"
          d="M115 115s4 2 5 0l14-14c3-2 6-3 8-3-8-11-15-24 2-41 5-5 10-7 16-7 1-2 3-7 12-11 0 0 5 3 7 16 4 2 8 5 12 9s7 8 9 12c14 3 17 7 17 7-4 8-9 11-11 11 0 6-2 11-7 16-16 16-30 10-41 2 0 3-1 7-5 11l-12 11c-1 1 1 5 1 5z"
        />
      </svg>
    </a>
  </div>
</div>
`;

exports[`Greeting renders 1`] = `
<div>
  <div
    class="sc-dkzDqf gcKDBg"
  >
    <h1>
       Start Shuffling! 
    </h1>
  </div>
</div>
`;
